### ISSUE TITLE
extension wrapper for the prosemirror-image-plugin

### ISSUE THREAD

**Issue by ebuychance** [ueberdosis/tiptap#2057]
### What problem are you facing?

More advanced Image functionality and tools.
This ProseMirror plugin meets all of my wants:
https://gitlab.com/emergence-engineering/prosemirror-image-plugin

Yet I am struggling to use this extension with TipTap.

### What’s the solution you would like to see?

An extension wrapper to consume the ProseMirror plugin.

### What alternatives did you consider?

I started writing an extension - but a little confused about about the schema, seems the plugin needs the editor.schema instance. 
So maybe it just needs to extend the schema, but I could not find any documentation on the extendNodeSchema extensions create option.

```
import { Extension } from '@tiptap/core';
import {
    defaultSettings,
    updateImageNode,
    imagePlugin,
} from 'prosemirror-image-plugin';
import { Schema } from 'prosemirror-schema-basic';

const imageSettings = { ...defaultSettings };

const imageSchema = new Schema({
    nodes: updateImageNode(Schema.spec.nodes, {
        ...imageSettings,
    }),
    marks: Schema.spec.marks,
});

export const ImageTools = Extension.create({
    name: 'image-tools',

    defaultOptions: { ...defaultSettings },

    addProseMirrorPlugins() {
        return [imagePlugin(imageSchema, { ...imageSettings })];
    },
});
```

### Anything to add? (optional)

thanks for any direction or insight in writing this extension.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by philippkuehn** [comment]
Oh, this was a fun challenge. Happy to see that it works fine together with tiptap. 
Here you go: https://codesandbox.io/s/tiptap-prosemirror-image-plugin-oddxs?file=/src/App.vue

**Comment by ashramsey** [comment]
Hey @philippkuehn I'm trying out the prose-mirror-plugin with react as per above linked setup. I'm getting a `RangeError
Index 3 out of range for <heading("ProseMirror image plugin"), paragraph("Oh, this image is resizable."), image>` when I use it along with a custom react component.

So I tried out your vuejs codesandbox you linked above, and could reproduce it with the Tiptap example docs vuejs extension component too. 

https://codesandbox.io/s/tiptap-prosemirror-image-plugin-forked-cnhgmg?file=/src/App.vue

I traced it down to [this bit of code](https://github.com/ProseMirror/prosemirror-model/blob/b99fa7246727a3a6a0c39896e6f2b702c650ea7b/src/node.ts#L175) that receives an undefined pos arg. I cant easily tell why.

Thanks
